build.macos32x86/common/Makefile.plugin

```diff
@@ -184,6 +196,21 @@ $(PLUGINPLIST): $(PLATDIR)/iOS/plugins/Info.plist getversion
 	sed "s/\$$(VERSION)/`./getversion VERSION_TAG`/" $< | \
 	sed "s/\$(LIBNAME)/$(LIBNAME)/" > $(PLUGINPLIST)
 
+$(PLUGINDYLIB): $(PLUGINREQS) $(VMDIR) $(OBJDIR) $(LIBOBJ)
+ifneq ($(LINK_WITH_CPP),)
+	$(LDCXX) \
+	$(DYFLAGS) \
+	-install_name @executable_path/Plugins/$(LIBNAME).dylib \
```

Comment:
ok, I will work on this, but please… do not stop to accept this PR because of this… I need VM to be built so Eliot can review some problems in callbacks and ephemerons (ortogonal problems, but they arise under certain circumstances in pharo images)

I can add rpath later :)

Esteban

> On 12 Sep 2016, at 20:34, Tobias Pape <notifications@github.com> wrote:
> 
> In build.macos32x86/common/Makefile.plugin <https://github.com/OpenSmalltalk/opensmalltalk-vm/pull/53#discussion_r78428837>:
> 
> > @@ -184,6 +196,21 @@ $(PLUGINPLIST): $(PLATDIR)/iOS/plugins/Info.plist getversion
> >  	sed "s/\$$(VERSION)/`./getversion VERSION_TAG`/" $< | \
> >  	sed "s/\$(LIBNAME)/$(LIBNAME)/" > $(PLUGINPLIST)
> >  
> > +$(PLUGINDYLIB): $(PLUGINREQS) $(VMDIR) $(OBJDIR) $(LIBOBJ)
> > +ifneq ($(LINK_WITH_CPP),)
> > +	$(LDCXX) \
> > +	$(DYFLAGS) \
> > +	-install_name @executable_path/Plugins/$(LIBNAME).dylib \
> It is about not using @executable_path, because that depends on the executable, notably in the Cocoa.app/Contents/MacOS/... path. But rather @rpath, which has a default List, which can be amended more extensibly. Quote from Apple:
> 
> To create a run-path dependent library, you specify a run-path–relative pathname as the library’s install name. A run-path-relative pathname uses the @rpath macro to specify a path relative to a directory to be determined at runtime. A run-path–relative pathname uses the following format:
> 
>  @rpath/<path_to_dynamic_library>
> These are examples of run-path–relative pathnames:
> 
> @rpath/libMyLib.dylib
> @rpath/MyFramework.framework/Versions/A/MyFramework A run-path install name is an install name that uses a run-path–relative pathname. You specify a run-path install name while creating the dependent library using the gcc -install_name option. See the gcc man page for more information.
> It is important to then install_name_tool -add_rpath @executable_path to the thing (Main executable) that loads the lib. (Picking a random example: PyPy. https://bitbucket.org/pypy/pypy/commits/3fa7727c46a3 <https://bitbucket.org/pypy/pypy/commits/3fa7727c46a3> )
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/OpenSmalltalk/opensmalltalk-vm/pull/53/files/e9feb5691e8545139768d3d4af51c37fb51ba9f4#r78428837>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AAfWXthjMj3o7hCDyGW5koaVb26fTQ_Vks5qpZs0gaJpZM4J61Et>.
>